### PR TITLE
TG-142: admin validation and flow

### PR DESCRIPTION
**Pull request by ilcosme**
This is a request to merge to master branch. We worked on implementing firebase to the admin resource to validate credentials and develope a flow according to what was validated. Also we added a redirect to home, and then according to what credentials are in use, it should display a flow to employees or to admins.

### CODE REVIEW

File: src/Components/AdminFlow/Profile/Profile.jsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    register,
+    formState: { errors },
+    reset
+  } = useForm({
+    mode: 'onBlur',
+    resolver: joiResolver(schema)
+  });
+  const UpdateAdmin = (data) => {
+    console.log(data);
+    // const admin = { ...data, _id: param.id };
```
Comment (nicolascostanza) [inline src/Components/AdminFlow/Profile/Profile.jsx:50]: borrar los console.log y comments

File: src/Components/AdminFlow/Profile/Profile.jsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import { appendErrors, useForm } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
+import { schema } from 'Components/AdminFlow/validations';
+//import { employeeValidationUpdate } from 'Components/EmployeesFlow/validations';
+import styles from './profile.module.css';
+import Modal from 'Components/Shared/Modal';
+import Sidebar from 'Components/Shared/Sidebar';
+
+function Profile() {
+  //const history = useHistory();
```
Comment (nicolascostanza) [inline src/Components/AdminFlow/Profile/Profile.jsx:15]: fuera comment

File: src/Components/AdminFlow/Profile/Profile.jsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                <input
+                  className={styles.inputsProfile}
+                  type="checkbox"
+                  name="active"
+                  {...register('active')}
+                  disabled={update}
+                  placeholder={adminSelected[0]?.active}
+                  error={appendErrors.active?.message}
+                ></input>
+                {/* {errors.active && <p className={styles.errorInput}>{errors.active?.message}</p>} */}
```
Comment (nicolascostanza) [inline src/Components/AdminFlow/Profile/Profile.jsx:196]: este igual

File: src/Components/AdminFlow/validations.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+//   password: Joi.string()
+//     .required('This field is required')
+//     .min(8)
+//     .regex(/(?!^[0-9]*$)(?!^[a-zA-Z]*$)^([a-zA-Z0-9]{8,25})$/)
+//     .messages({
+//       'string.min': 'Password must contain at least 8 characters',
+//       'string.pattern.base': 'Password must contain letters and numbers',
+//       'string.empty': 'This field is required'
+//     })
+// });
```
Comment (nicolascostanza) [inline src/Components/AdminFlow/validations.js:156]: borrar

File: src/Components/EmployeesFlow/Profile/Profile.jsx [hunk 5]
```diff
@@ -50,6 +50,7 @@ function Profile() {
     resolver: joiResolver(employeeValidationUpdate)
   });
   const UpdateEmployee = (data) => {
+    console.log(data);
```
Comment (nicolascostanza) [inline src/Components/EmployeesFlow/Profile/Profile.jsx:53]: borrar

File: src/Components/Home/index.js [hunk 6]
```diff
@@ -1,13 +1,49 @@
...
 function Home() {
-  return (
-    <section className={styles.container}>
-      <Sidebar></Sidebar>
-      <h2>Home</h2>
-    </section>
-  );
+  const history = useHistory();
+  const role = useSelector((state) => state.auth.authenticated?.role);
+  console.log(role);
```
Comment (nicolascostanza) [inline src/Components/Home/index.js:10]: borrar

File: src/Components/AdminFlow/Profile/Profile.jsx (hunk 1 above)
Comment (martinalbrnz) [inline src/Components/AdminFlow/Profile/Profile.jsx:50]: ojo el clog y el comment

File: src/Components/Layout/index.js [hunk 7]
```diff
@@ -28,13 +28,23 @@ function Layout() {
             <Route exact path="/" component={Home} />
             <Route exact path="/signup" component={SignUp} />
             <Route exact path="/login" component={Login} />
-            <Redirect to="/login" />
             <PrivateRoute exact path="/admins" role="ADMIN" component={Admins} />
-            <PrivateRoute exact path="/super-admins" role="SUPERADMIN" component={SuperAdmins} />
+            <PrivateRoute
+              exact
+              path="/admins/profile/:id"
+              role={'ADMIN'}
```
Comment (martinalbrnz) [inline src/Components/Layout/index.js:35]: role={['ADMIN']}
Just to mantain consistency

**Review by nicolascostanza — APPROVED** [review]

File: src/Components/Shared/Sidebar/index.jsx [hunk 8]
```diff
@@ -25,7 +25,7 @@ const Sidebar = ({ children }) => {
         </div>
         <a>Home</a> {/*<Link to="/tasks">home</Link>*/}
```
Comment (martinalbrnz) [inline src/Components/Shared/Sidebar/index.jsx:26]: `<a onClick={() => history.push('/')}>Home</a>`
i think this could help

**Review by martinalbrnz — APPROVED** [review]
Good Job!

**Comment by RominaRIP** [comment]
## Testing results

Fer, please check the following bug report

Tested locally on https://alfonso-trackgenix-app-git-tg-142-basp-m2022.vercel.app/
Test Results: Failed ⚠️

Bugs found:

- (1) The 'Admin' role shouldn't have permission to edit/delete Admins
- (2) The 'Employee' role shouldn't have permissions to edit/delete Employees, Projects or Tasks
- (3) The modal displayed after the login action is empty:

empty-modal-bug

- (4) There are two success messages displayed when the Admin updates his/her profile:

first-prompt
second-success-msg

- (5) The modal displayed when the Admin updates an employee is not well aligned:

modal-not-aligned

**Review by nicolascostanza — APPROVED** [review]

**Review by martinalbrnz — APPROVED** [review]
LGTM

**Pull request merged**